src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -217,46 +227,52 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_aggregate_node(
     auto current_column_id = static_cast<ColumnID::base_type>(groupby_columns.size());
 
     for (auto& aggregate_expression : aggregate_expressions) {
-      Assert(aggregate_expression->expression_list().size(), "Aggregate: empty expression list");
+      Assert(aggregate_expression->aggregate_function_arguments().size(), "Aggregate: empty expression list");
       DebugAssert(aggregate_expression->type() == ExpressionType::Function, "Expression is not a function.");
 
       // Do not project for COUNT(*)
       if (aggregate_expression->aggregate_function() == AggregateFunction::Count &&
-          (aggregate_expression->expression_list())[0]->type() == ExpressionType::Star) {
+          (aggregate_expression->aggregate_function_arguments())[0]->type() == ExpressionType::Star) {
         continue;
       }
 
       // Add original expression of the function to the Projection.
-      column_expressions.emplace_back(aggregate_expression->expression_list()[0]);
+      Assert(aggregate_expression->aggregate_function_arguments().size() == 1,
+             "Aggregate functions with more than one argument not supported right now");
+      const auto argument_lqp_expression =
+          std::dynamic_pointer_cast<OperatorExpression>(aggregate_expression->aggregate_function_arguments()[0]);
+      DebugAssert(argument_lqp_expression, "Expression in AggregateNode was not an LQPExpression");
```

Comment:
I think we always use present tense in error messages

Reply:
done